gh-123090: Add the `P` format character (#123097)
python/cpython: Doc/c-api/arg.rst

   .. versionadded:: 3.3

> ``p*`` (:class:`bool`) [_Bool]

Reviewer: ```suggestion
``P`` (:class:`bool`) [_Bool]
```

>    Like ``p``, but the C pointer should be a ``_Bool``/``bool`` instead of an integer.

Reviewer 2: ```suggestion
   Like ``p``, but converts to a :c:expr:`_Bool` instead of an integer.
```
